examples/bri-3/src/bri/transactions/agents/transactions.agent.ts

```diff
@@ -0,0 +1,90 @@
+import {
+  BadRequestException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common';
+import { Transaction } from '../models/transaction';
+import { TransactionStatus } from '../models/transactionStatus.enum';
+
+import {
+  DELETE_WRONG_STATUS_ERR_MESSAGE,
+  NOT_FOUND_ERR_MESSAGE,
+  UPDATE_WRONG_STATUS_ERR_MESSAGE,
+} from '../api/err.messages';
+import { TransactionStorageAgent } from './transactionStorage.agent';
+import { BpiAccount } from '../../identity/bpiAccounts/models/bpiAccount';
+
+@Injectable()
+export class TransactionAgent {
+  constructor(private repo: TransactionStorageAgent) {}
+  public throwIfCreateTransactionInputInvalid() {
+    return true;
+  }
+
+  public createNewTransaction(
+    transactionId: string,
+    nonce: number,
+    workflowInstanceId: string,
+    workstepInstanceId: string,
+    from: BpiAccount,
+    to: BpiAccount,
+    payload: string,
+    signature: string,
+    status: TransactionStatus,
+  ): Transaction {
+    return new Transaction(
+      transactionId,
+      nonce,
+      workflowInstanceId,
+      workstepInstanceId,
+      from,
+      to,
+      payload,
+      signature,
+      status,
+    );
+  }
+
+  public async fetchUpdateCandidateAndThrowIfUpdateValidationFails(
+    id: string,
+  ): Promise<Transaction> {
+    const transactionToUpdate = await this.repo.getTransactionById(id);
+
+    if (!transactionToUpdate) {
+      throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
+    }
+
+    if (transactionToUpdate.status != TransactionStatus.Initialized) {
```

Comment:
similar validation for create?

Reply:
it will be hardcoded so no reason to validate status.